lib/rspec/matchers/built_in/yield.rb

```diff
@@ -182,35 +182,44 @@ def supports_block_expectations?
       private
 
         def set_expected_yields_count(relativity, n)
+          raise NotImplementedError, "Multiple count constraints are not supported" if @expectation_type
```

Comment:
```suggestion
          raise ArgumentError, "Multiple constraints are not currently supported" if @expectation_type
```

Reply:
It's definitely *not* and `ArgumentError`, i.e. there are no possible arguments that would not generate this error and there might not be arguments at all. Some cases will also never be implemented and always be an error (e.g. `include(...).twice.once`). Maybe just a `RuntimeError`?

Reply:
Its an `ArgumentError` because it is attempting to overwrite existing arguments.

Reply:
I don't feel that strongly about it though, you can change to a `StandardError` for now, and I hope to find some time to fix it properly, in which case `.once.twice` etc will raise an `ArgumentError` (because thats an invalid combination) whilst `at_least(:once).at_most(:thrice)` would work.

Reply:
Let's avoid an argument and agree to disagree on what is an argument 😅

I believe it is always a mistake to raise a (naked) `StandardError`. I'm not aware of Ruby ever raising one for example. That class is meant for rescuing only.

I've changed it to a `RuntimeError`.